File: app/src/main/java/dev/ashish/newsapp/viewmodel/NewsViewModelFactory.java
package dev.ashish.newsapp.viewmodel;

import android.app.Application;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.annotation.NonNull;

public class NewsViewModelFactory implements ViewModelProvider.Factory {

    private Application mApplication;
    private String mParam;

    public NewsViewModelFactory(Application application, String taskId) {
        mApplication = application;
        mParam = taskId;
    }


    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        //noinspection unchecked
        return (T) new MyViewModel( mApplication,mParam);
    }
}

File: app/src/main/java/dev/ashish/newsapp/viewmodel/NewsViewModelFactory.kt
package dev.ashish.newsapp.viewmodel

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class NewsViewModelFactory(private val mApplication: Application, private val mParam: String?) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return MyViewModel(mApplication, mParam) as T
    }

}